script.js

```diff
@@ -1,4 +1,138 @@
+  if (playerMode == "Player 1") {
+    var randomDiceRoll = diceRollArray();
+    console.log("Two dices rolled by player 1", randomDiceRoll);
+    myOutputValue = `Welcome Player 1. <br> <br> You rolled ${randomDiceRoll[0]} for Dice 1 and ${randomDiceRoll[1]} for Dice 2. <br> <br> Choose the order of the dice.`;
+
+    playerMode = "Choose order for player 1";
+    return myOutputValue;
+  }
+
+  //after the game goes into choose order mode, if player 1 choose dice 1, it will show array[0] + array[1]
+  //if player 1 choose dice 2, it will show array[1] + array[0]
+  //the game will go into player 2 mode
+
+  if (playerMode == "Choose order for player 1" && playerInput == 1) {
+    var player1ArrayInNumber = Number(numPlayer1DiceArray(1));
+    console.log("Player 1 choose dice 1", player1ArrayInNumber);
+
+    myOutputValue = `Player 1, you chose Dice ${playerInput} first. <br> <br> Your number is ${player1ArrayInNumber} <br> <br> It's now Player 2's turn.`;
+    playerMode = "Player 2";
+    return myOutputValue;
+  } else if (playerMode == "Choose order for player 1" && playerInput == 2) {
+    var player1ArrayInNumber = Number(numPlayer1DiceArray(2));
+    console.log("Player 1 choose dice 2", player1ArrayInNumber);
+    myOutputValue = `Player 1, you chose Dice ${playerInput} first. <br> <br> Your number is ${player1ArrayInNumber}.  <br> <br> It's now Player 2's turn.`;
+    playerMode = "Player 2";
+    return myOutputValue;
+  }
+
+  //if it's player 2's turn to play the game, he will roll the dice and generate 2 dice rolls
+  if (playerMode == "Player 2") {
+    var randomDiceRoll = diceRollArray();
+    console.log("Two dices rolled by player 2", randomDiceRoll);
+    myOutputValue = `Welcome Player 2. <br> <br> You rolled ${randomDiceRoll[0]} for Dice 1 and ${randomDiceRoll[1]} for Dice 2. <br> <br> Choose the order of the dice.`;
+
+    playerMode = "Choose order for player 2";
+    return myOutputValue;
+  }
+
+  //after the game goes into choose order mode, if player 2 choose dice 1, it will show array[0] + array[1]
+  //the game will show whether player 1 or 2 wins, depends on which number is larger
+  //depending on player 1 and 2's number, the larger number wins.
+  //game goes back to "Player 1" mode
+
+  if (playerMode == "Choose order for player 2" && playerInput == 1) {
+    var player2ArrayInNumber = Number(numPlayer2DiceArray(1));
+    console.log("Player 2 choose dice 1", player2ArrayInNumber);
+    myOutputValue = `Player 2, you chose Dice ${playerInput} first. <br> <br> Your number is ${player2ArrayInNumber}. `;
+
+    if (
+      Number(numPlayer1DiceArray(1)) < Number(numPlayer2DiceArray(1)) ||
+      Number(numPlayer1DiceArray(2)) < Number(numPlayer2DiceArray(1))
+    ) {
+      var message = `You win`;
+    } else if (
+      Number(numPlayer1DiceArray(1)) > Number(numPlayer2DiceArray(1)) ||
+      Number(numPlayer1DiceArray(2)) > Number(numPlayer2DiceArray(1))
```

Comment:
Mentioned this to you during inclass last week - but yeah its better to get the players to decide their number before comparing - this will especially be true for a game like blackjack where the player has more choices to make and so trying to determine the outcome beforehand is near impossible